src/OptionResult.ts

```diff
@@ -589,6 +600,17 @@ class __Result<A, E> {
     return this.tag === "Ok" ? this.value : defaultValue;
   }
 
+  /**
+   * Return the value if present, otherwise throw an error
+   */
+  getOrThrow(this: Result<A, E>): A | never {
+    if (this.tag === "Error") {
+      throw new Error("Tried to unwrap a Result.Error value");
```

Comment:
I'd throw `this.error` directly in this case, or at least put it in the error's `cause`.

Reply:
Nice idea, I wanted to use `cause`, unfortunately it is not supported yet by the Typescript version used in the project.

Now throwing `this.error` if the error is string. Also added the implementation for AsyncData